File: 020.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st = stack<char>(); 
        char aux; 
        for (char c: s){
            if (c == '(') st.push(')'); 
            else if (c == '[') st.push(']');
            else if (c == '{') st.push('}');
            else {
                if (st.empty()) return false; 
                aux = st.top(); 
                st.pop(); 
                if (aux != c) return false; 
            }
        }
        return st.empty(); 
    }

};

File: 020.py
class Solution:
    def isValid(self, s: str) -> bool:
        q = []
        
        for i in s: 
            if i == "(":
                q.append(")") 
            elif i == "[":
                q.append("]")
            elif i == "{":
                q.append("}") 
            elif q == [] or q.pop() != i:
                return False 
                
        return len(q) == 0
